src/common/components/DateInfo.tsx

```diff
@@ -22,8 +26,45 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-const DateInfo = ({ date }: DateInfoProps): JSX.Element => {
+const DateInfo = ({ resource }: DateInfoProps): JSX.Element => {
   const classes = useStyles();
+  const date = useMemo(() => {
+    const dateOrPeriod = getResourceDateOrPeriod(resource);
+    if (typeof dateOrPeriod === "string") {
```

Comment:
is this the best we can do in of typing ?
in particular, i'm not a fan of `if (typeof dateOrPeriod == "object")`

Reply:
I can't really come up with a better solution to check whether the result is a `string` of `{ start: string, end: string }`

https://stackoverflow.com/questions/8511281/check-if-a-value-is-an-object-in-javascript

Reply:
ok dommage